tt_metal/impl/device/device.cpp

```diff
@@ -3229,6 +3229,15 @@ void Device::set_worker_mode(const WorkExecutorMode& mode) {
 void Device::enable_async(bool enable) {
     auto mode = enable ? WorkExecutorMode::ASYNCHRONOUS : WorkExecutorMode::SYNCHRONOUS;
     this->set_worker_mode(mode);
+    // If a worker thread is spawned for a device, register/track it in a runtime structure.
+    // If a worker thread is destroyed, remove it from the structure.
+    // This is required for checking if a call is made from an application thread or a worker thread.
+    // See InApplicationThread().
+    if (enable) {
+        tt::DevicePool::instance().register_worker_thread_for_device(this, this->work_executor.get_worker_thread_id());
+    } else {
+        tt::DevicePool::instance().delete_worker_thread_for_device(this);
```

Comment:
There is an asymmetry here. When enabled you `register`, but when disabled, you are not unregistering, but `deleting`. Likely just a naming

Reply:
Done